nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/settings/ui/flow-analysis-rules/flow-analysis-rule-table/flow-analysis-rule-table.component.ts

```diff
@@ -0,0 +1,265 @@
+            } else if (entity.status.runStatus === 'ENABLED') {
+                return 'enabled fa fa-flash';
+            }
+        }
+        return '';
+    }
+
+    formatState(entity: FlowAnalysisRuleEntity): string {
+        if (entity.status.validationStatus === 'VALIDATING') {
+            return 'Validating';
+        } else if (entity.status.validationStatus === 'INVALID') {
+            return 'Invalid';
+        } else {
+            if (entity.status.runStatus === 'DISABLED') {
+                return 'Disabled';
+            } else if (entity.status.runStatus === 'ENABLED') {
+                return 'Enabled';
+            }
+        }
+        return '';
+    }
+
+    formatType(entity: FlowAnalysisRuleEntity): string {
+        return this.nifiCommon.formatType(entity.component);
+    }
+
+    formatBundle(entity: FlowAnalysisRuleEntity): string {
+        return this.nifiCommon.formatBundle(entity.component.bundle);
+    }
+
+    isDisabled(entity: FlowAnalysisRuleEntity): boolean {
+        return entity.status.runStatus === 'DISABLED';
+    }
+
+    isEnabledOrEnabling(entity: FlowAnalysisRuleEntity): boolean {
+        return entity.status.runStatus === 'ENABLED' || entity.status.runStatus === 'ENABLING';
+    }
+
+    hasActiveThreads(entity: ReportingTaskEntity): boolean {
+        return entity.status?.activeThreadCount > 0;
+    }
+
+    canConfigure(entity: FlowAnalysisRuleEntity): boolean {
+        return this.canRead(entity) && this.canWrite(entity) && this.isDisabled(entity);
+    }
+
+    configureClicked(entity: FlowAnalysisRuleEntity, event: MouseEvent): void {
+        event.stopPropagation();
+        this.configureFlowAnalysisRule.next(entity);
+    }
+
+    canEnable(entity: FlowAnalysisRuleEntity): boolean {
+        const userAuthorized: boolean = this.canRead(entity) && this.canOperate(entity);
+        return userAuthorized && this.isDisabled(entity) && entity.status.validationStatus === 'VALID';
+    }
+
+    enabledClicked(entity: FlowAnalysisRuleEntity, event: MouseEvent): void {
+        this.enableFlowAnalysisRule.next(entity);
+    }
+
+    canDisable(entity: FlowAnalysisRuleEntity): boolean {
+        const userAuthorized: boolean = this.canRead(entity) && this.canOperate(entity);
+        return userAuthorized && this.isEnabledOrEnabling(entity);
+    }
+
+    disableClicked(entity: FlowAnalysisRuleEntity, event: MouseEvent): void {
+        this.disableFlowAnalysisRule.next(entity);
+    }
+
+    canChangeVersion(entity: FlowAnalysisRuleEntity): boolean {
+        return (
+            this.isDisabled(entity) &&
+            this.canRead(entity) &&
+            this.canWrite(entity) &&
+            entity.component.multipleVersionsAvailable === true
+        );
+    }
+
+    canDelete(entity: FlowAnalysisRuleEntity): boolean {
+        return this.isDisabled(entity) && this.canRead(entity) && this.canWrite(entity) && this.canModifyParent(entity);
```

Comment:
We can check directly check the users `controller` permissions. We don't need to use a callback approach here like we needed with Controller Services since their parents depend on its scope.